interfaces/cython/cantera/_cantera.pxd

```diff
@@ -187,6 +191,7 @@ cdef extern from "cantera/thermo/ThermoPhase.h" namespace "Cantera":
         string type()
         string phaseOfMatter() except +translate_exception
         void getSpeciesParameters(string, CxxAnyMap&) except +translate_exception
+        CxxAnyMap& input()
```

Comment:
This maybe is a little late to catch this, but why do some classes use a getter method while others just expose the `AnyMap` directly?

Reply:
I actually had the same question when this was first introduced (and posted as much). If I remember correctly, in some instances, the `AnyMap` holds static information and is accessed directly. In other instances, it needs to be assembled, and thus uses a getter.

Reply:
The `input()` getter is just a simple getter in all cases where it's used -- the `getParameters(AnyMap&)` or `AnyMap parameters()` methods are the ones that do work to determine their output.

The cases where `input` is just a public member variable arose in classes that were (at least originally) basically data containers where all of the members were public and there were few if any member functions. The `GasTransportData` class still mostly follows this model, for example. For other classes that don't expose any public member data, e.g. `ThermoPhase`, the getter function is used.

I think debating whether the classes with public data members were a good idea is a ship that sailed a while ago. I'm pretty sure the `input` member for most if not all of these is already present as part of the 2.5 release.

Reply:
Thanks for reiterating that this dates back to before the recent addition of YAML serialization. I agree that this is now de facto part of the released API.

Reply:
Although this has been released already, I think considering whether we can improve the consistency of the API is worthwhile, at least briefly. My concern is that anyone new coming into this code (myself, for example) will look at this and have no idea why some classes have public member data and some have getters, and since there's not really an apparent pattern as to which classes have which, it makes understanding the code that much more difficult. I'm trying to keep in mind that small inconsistencies and gotchas like this are things that we have to teach to the next generation, and there's already more than enough of that to go around 😄 

The simplest solution is probably to add getter methods to classes with public member data and prefer using those, even if the member data remains public. That way we can document the behavior as semi-intentional and have consistence of interfaces. I'm not sure if this would be consistent with C++ coding style, though (I imagine not).

Reply:
FWIW, the coding guide does not disallow public member variables, as it states: "Protected and private member variable names are generally prefixed with `m_`. For most classes, member variables should not be public."

The way I view it, `input` (as getter or member variable), is mostly for internal use while the `getParameters(AnyMap&)` or `AnyMap parameters()` methods provide a consistent interface. From that perspective, adding the clarification `@internal` may sufficiently resolve this controversy. Having multiple interfaces for a rarely used method is imho not necessary (neither is the significant effort for deprecation of a member variable, which is a major undertaking).

Reply:
OK, this has evolved into much more discussion than I intended. I think at this point, the conversation has resolved itself and no changes to the code are needed in this PR. In the future, we may consider some way to resolve the consistency issue that I see here, to avoid confusing future developers, since as I said earlier, there's no pattern that is obvious to me which classes have direct access and which use a getter.

Reply:
> If you want to see how difficult it is to change the _type_ of an existing member (which is roughly equivalent to replacing a public member variable into a getter)

Just to be clear, I'm not suggesting _replacing_ the public member variable with a getter, but adding a getter _in addition_ to the public access, with a consistent and documented preference for using the getter _within Cantera's code_. There are (at least) 2 caveats to that: 1) I'm not sure whether the compiler would allow that and 2) I'm not sure how bad a violation (if at all) of implicit C++ code style guides it would be (that is, if an experienced C++ programmer looked at that, would they insist on removing it). I know that this is suboptimal, in terms of wanting to have one way to do things. However, I think it would be a benefit for Cantera developers (us, future us, and other future developers) to have a consistent way to access this data, since there isn't an obvious (to me) pattern that governs when a public member is used vs. when a getter is used.